### ISSUE TITLE
Cannot find cyipopt module after installation

### ISSUE THREAD

**Issue by HuaweiWang** [matthias-k/cyipopt#21]
Hi,

I was working with Linux and this Ipopt interface works very well. Recently, I am trying to install Ipopt optimization system with Anaconda python on a Windows Server. The Issue I got is that cyipopt module cannot be found after installed everything.

Here is my installation process:

First, I installed Anaconda-64bit of python 2.7, and the necessary packages for cyipopt.

Then, I download [Ipopt binaries (3.11.0)](http://www.coin-or.org/download/binary/Ipopt/), and put them in the corresponding location. 

Finally, I changed the PATH in cyipopt _setup.py_ and installed it using _python setup.py install_ 

Here is the terminal information while executing cyipopt setup.py.

```
_(base) C:\cyipopt>python setup.py install
C:\Anaconda2\lib\site-packages\setuptools\dist.py:388: UserWarning: Normalizing
'0.1.8.dev' to '0.1.8.dev0'
  normalized_version,
running install
running bdist_egg
running egg_info
writing requirements to ipopt.egg-info\requires.txt
writing ipopt.egg-info\PKG-INFO
writing top-level names to ipopt.egg-info\top_level.txt
writing dependency_links to ipopt.egg-info\dependency_links.txt
reading manifest file 'ipopt.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'include*'
writing manifest file 'ipopt.egg-info\SOURCES.txt'
installing library code to build\bdist.win-amd64\egg
running install_lib
running build_py
running build_ext
skipping 'src\cyipopt.c' Cython extension (up-to-date)
creating build\bdist.win-amd64\egg
creating build\bdist.win-amd64\egg\ipopt
copying build\lib.win-amd64-2.7\ipopt\cyipopt.pyd -> build\bdist.win-amd64\egg\i
popt
copying build\lib.win-amd64-2.7\ipopt\ipopt_wrapper.py -> build\bdist.win-amd64\
egg\ipopt
copying build\lib.win-amd64-2.7\ipopt\version.py -> build\bdist.win-amd64\egg\ip
opt
copying build\lib.win-amd64-2.7\ipopt\__init__.py -> build\bdist.win-amd64\egg\i
popt
byte-compiling build\bdist.win-amd64\egg\ipopt\ipopt_wrapper.py to ipopt_wrapper
.pyc
byte-compiling build\bdist.win-amd64\egg\ipopt\version.py to version.pyc
byte-compiling build\bdist.win-amd64\egg\ipopt\__init__.py to __init__.pyc
creating stub loader for ipopt\cyipopt.pyd
byte-compiling build\bdist.win-amd64\egg\ipopt\cyipopt.py to cyipopt.pyc
installing package data to build\bdist.win-amd64\egg
running install_data
copying lib_mt\x64\ReleaseMKL\IpOpt-vc8.dll -> build\bdist.win-amd64\egg\ipopt
copying lib_mt\x64\ReleaseMKL\IpOpt-vc10.dll -> build\bdist.win-amd64\egg\ipopt
copying lib_mt\x64\ReleaseMKL\IpOptFSS.dll -> build\bdist.win-amd64\egg\ipopt
creating build\bdist.win-amd64\egg\EGG-INFO
copying ipopt.egg-info\PKG-INFO -> build\bdist.win-amd64\egg\EGG-INFO
copying ipopt.egg-info\SOURCES.txt -> build\bdist.win-amd64\egg\EGG-INFO
copying ipopt.egg-info\dependency_links.txt -> build\bdist.win-amd64\egg\EGG-INF
O
copying ipopt.egg-info\requires.txt -> build\bdist.win-amd64\egg\EGG-INFO
copying ipopt.egg-info\top_level.txt -> build\bdist.win-amd64\egg\EGG-INFO
writing build\bdist.win-amd64\egg\EGG-INFO\native_libs.txt
zip_safe flag not set; analyzing archive contents...
creating 'dist\ipopt-0.1.8.dev0-py2.7-win-amd64.egg' and adding 'build\bdist.win
-amd64\egg' to it
removing 'build\bdist.win-amd64\egg' (and everything under it)
Processing ipopt-0.1.8.dev0-py2.7-win-amd64.egg
Removing c:\anaconda2\lib\site-packages\ipopt-0.1.8.dev0-py2.7-win-amd64.egg
Copying ipopt-0.1.8.dev0-py2.7-win-amd64.egg to c:\anaconda2\lib\site-packages
ipopt 0.1.8.dev0 is already the active version in easy-install.pth

Installed c:\anaconda2\lib\site-packages\ipopt-0.1.8.dev0-py2.7-win-amd64.egg
Processing dependencies for ipopt==0.1.8.dev0
Searching for setuptools==39.1.0
Best match: setuptools 39.1.0
Adding setuptools 39.1.0 to easy-install.pth file
Installing easy_install-script.py script to C:\Anaconda2\Scripts
Installing easy_install.exe script to C:\Anaconda2\Scripts

Using c:\anaconda2\lib\site-packages
Searching for future==0.16.0
Best match: future 0.16.0
Adding future 0.16.0 to easy-install.pth file

Using c:\anaconda2\lib\site-packages
Searching for six==1.11.0
Best match: six 1.11.0
Adding six 1.11.0 to easy-install.pth file

Using c:\anaconda2\lib\site-packages
Searching for Cython==0.28.2
Best match: Cython 0.28.2
Adding Cython 0.28.2 to easy-install.pth file
Installing cython-script.py script to C:\Anaconda2\Scripts
Installing cython.exe script to C:\Anaconda2\Scripts
Installing cygdb-script.py script to C:\Anaconda2\Scripts
Installing cygdb.exe script to C:\Anaconda2\Scripts
Installing cythonize-script.py script to C:\Anaconda2\Scripts
Installing cythonize.exe script to C:\Anaconda2\Scripts

Using c:\anaconda2\lib\site-packages
Searching for scipy==1.1.0
Best match: scipy 1.1.0
Adding scipy 1.1.0 to easy-install.pth file

Using c:\anaconda2\lib\site-packages
Searching for numpy==1.14.3
Best match: numpy 1.14.3
Adding numpy 1.14.3 to easy-install.pth file

Using c:\anaconda2\lib\site-packages
Finished processing dependencies for ipopt==0.1.8.dev0_
```
It seems only an .egg file was generated. To avoid this, I tried this: 
_python setup.py install --single-version-externally-managed --record=record.txt_

An Ipopt folder was created under Anaconda2\Lib\site-packages to include ipopt and cyipopt binaries.

However, When I was trying to run _examplehs071.py_ in both cases, it shows that no module named cyipopt.

I also found that someone had the same [issue ](https://stackoverflow.com/questions/39193008/how-to-alter-setup-py-for-cyipopt) one year ago, but haven't been answered.

Any suggestion that how can I solve this problem? Thanks in advance!

**Comment by moorepants** [comment]
I finally had a chance to try this on windows and I hit the same error. The binaries seems to be installed into the correct places, but I get the error `ImportError: DLL load failed: The specified module could not be found.`. Even when I change to the directory with the cyipopt pyd file, it fails to import.

**Comment by moorepants** [comment]
This could be a compiler inconsistency with the compiler conda-forge/anaconda uses for Cython/Python and the one we are using for the cyipopt compilation.

**Comment by moorepants** [comment]
Here is a another tidbit from https://github.com/cython/cython/wiki/InstallingOnWindows#using-anaconda:

> To use Cython via Anaconda, you should first install a 32-bit Anaconda. 64-bit Anaconda will also work, but you may encounter issues when you will try to compile your Cython extensions.

**Comment by moorepants** [comment]
We now have very specific instructions for installations with custom ipopt:

https://github.com/matthias-k/cyipopt#example-installation-on-ubuntu-1804-with-custom-compiled-ipopt

Please try this and raise new issues if you are still having trouble.

**Comment by moorepants** [comment]
Oops, I see this is windows.

**Comment by shuanghuabai** [comment]
I hit the same error. Installation is complete actually, the thing is, installed package is named as 'ipopt', however in the __init__.py you have the code 'import cyipopt ...'. It seems this auther haven't even made up his mind on the name of the package he is writing.

**Comment by hooyeh** [comment]
I have the same issue on windows: ModuleNotFoundError: No module named 'cyipopt'. 

I did the same thing as https://stackoverflow.com/questions/39193008/how-to-alter-setup-py-for-cyipopt to modify the setup.py file.

**Comment by apommel** [comment]
I managed to resolve this problem (on Windows) by replacing the lines "from cyipopt import *" by "from .cyipopt import *" in __init__.py and "import cyipopt" by "import ipopt.cyipopt as cyipopt" in ipopt_wrapper.py. All the tests seem to execute correctly.

**Comment by hooyeh** [comment]
> I managed to resolve this problem (on Windows) by replacing the lines "from cyipopt import *" by "from .cyipopt import *" in **init**.py and "import cyipopt" by "import ipopt.cyipopt as cyipopt" in ipopt_wrapper.py. All the tests seem to execute correctly, but I am not sure if it affects compatibility on Linux.

Works beautifully in Windows. Many thanks!

**Comment by moorepants** [comment]
@apommel Great! If you'd like to submit a PR we'd be happy to accept it.

**Comment by moorepants** [comment]
I think we have this resolved in #49.
